### PR TITLE
[FIX] grouping documents for unprivileged users

### PR DESCRIPTION
**Pull request by hbrunn**
Deep within `read_group`, there's a query that selects some record of the group, of which we don't have a lot of control. If this record doesn't happen to be readable by the user, you'll only get a read error when `read_group` tries to `read` this record.

So this patch restricts `read_group` to records the user has access to anyways. I'm aware that this patch is a catastrophe for performance, but it seems to me it's better to have it forbiddingly slow than not working. For ~10000 documents, it performs okay on my test system.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Is this still relevant for you?

**Comment by hbrunn** [comment]
yes

File: addons/document/document.py [hunk 1]
```diff
@@ -160,6 +160,27 @@ def __get_partner_id(self, cr, uid, res_model, res_id, context=None):
             return bro.partner_id.id
         return False
 
+    def read_group(self, cr, uid, domain, fields, groupby, offset=0,
+                   limit=None, context=None, orderby=False):
+        """ Gouping breaks for unprivileged users if the sql statement
```
Comment (pedrobaeza) [inline addons/document/document.py:165]: s/Gouping/Grouping

**Review by pedrobaeza — COMMENTED** [review]
Shouldn't this be considered at ORM level instead?

**Comment by hbrunn** [comment]
it's a result of how the document module manipulates access permissions in 7. We're phasing out 7 for our last customers, so I'm not going to touch this any more

**Comment by pedrobaeza** [comment]
OK!

**Pull request closed without merging**